src/ORM/DataObjectSchema.php

```diff
@@ -902,12 +920,36 @@ public function hasOneComponent($class, $component)
             return null;
         }
 
+        $spec = $hasOnes[$component];
+
         // Validate
-        $relationClass = $hasOnes[$component];
+        if (is_array($spec)) {
+            $this->checkHasOneArraySpec($class, $component, $spec);
+        }
+        $relationClass = is_array($spec) ? $spec['class'] : $spec;
```

Comment:
It isn't needed here because an exception will be thrown in `checkHasOneArraySpec` if it's missing.
It _is_ needed in the relation validation service because this validation isn't performed within that service.